influxdb3_write/src/persister.rs

```diff
@@ -22,26 +26,119 @@ impl PersisterImpl {
 #[async_trait]
 impl Persister for PersisterImpl {
     async fn load_catalog(&self) -> crate::Result<Option<PersistedCatalog>> {
-        todo!()
+        let mut list = self
+            .object_store
+            .list(Some(&CatalogFilePath::dir()))
+            .await?;
+        let mut catalog_path: Option<ObjPath> = None;
+        while let Some(item) = list.next().await {
+            let item = item?;
+            catalog_path = match catalog_path {
+                Some(old_path) => {
+                    let new_catalog_name = item
+                        .location
+                        .filename()
+                        .expect("catalog names are utf-8 encoded");
```

Comment:
I think the message can change to be "catalog names are guaranteed to exist" in this dir, but we should have a filename available to us given how we store them. If however we expect other things to be written there without a name then I think we can change this up to continue to the next item in the loop if it has no file name. In fact it might be better to just do that